### DISCUSSION TITLE
how do you process 50 batches before giving the loss output

### DISCUSSION THREAD

**Discussion by neurosynapse** [open-mmlab/mmdetection#9232] [Q&A]
Hello,

It would be nice if you could help me to understand how its implemented that you process 50 batches at once. Here is the example of retinanet training, batch size is 4 but the code loads 50 batches in one iteration and then gives the loss output. Is the loss calculated for each batch separately and then you get average at the end? How did you implement this training process as I see it greatly improves the training speed compared to single batch process code variants. Could you explain why? Its a little bit hard to debug the code because of the register usage. Thanks a lot!

Here I print the input shape for the model, as you can see it loads up 50 batches and then gives the loss:

![image](https://user-images.githubusercontent.com/47304445/199800334-dddd2795-179f-4d91-814f-d38e26b2875a.png)

Best regards,
Roberts

**Comment by nijkah** [comment]
Hi @Franko9999.

Actually, `mmdetection` and related frameworks process each batch as one step (e.g. `Model.forward()`, `loss.backward()`, `optimizer.step()`).
The logger saves related information in each step, and just shows averaged metrics when the `interval` is met. https://github.com/open-mmlab/mmdetection/blob/e71b499608e9c3ccd4211e7c815fa20eeedf18a2/configs/_base_/default_runtime.py#L4

It would be nice to check [`Runner`](https://github.com/open-mmlab/mmcv/blob/94c071b310/mmcv/runner/iter_based_runner.py) in mmcv if you are curious about its detailed logic.

**Reply by neurosynapse** [reply]
Hi, 

Yes, I see. Thanks for the reply. Upvoted.

Best regards,
Roberts
